lightningd/gossip_control.c

```diff
@@ -66,15 +66,19 @@ static void got_txout(struct bitcoind *bitcoind,
 		      struct short_channel_id *scid)
 {
 	const u8 *script;
+	u64 satoshis = 0;
 
 	/* output will be NULL if it wasn't found */
-	if (output)
+	if (output) {
 		script = output->script;
-	else
+		satoshis = output->amount;
+	} else {
 		script = NULL;
+	}
```

Comment:
I have a minor preference for setting things to NULL in an if branch, over initialization.  That way the compiler will (sometimes) warn you when it's used unset, if anything is missed.  See how script is done here, for example.